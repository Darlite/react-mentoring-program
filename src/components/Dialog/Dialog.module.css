.dialogContainer {
    display: flex;
    flex-direction: column;
    position: absolute;
    top: 0;
    left: 50%;
    transform: translate(-488px, 100px);
    width: 976px;
    background: #232323;
    padding: 40px;
    color: #FFFFFF;
}

.closeButton {
    position: absolute;
    top: 20px;
    right: 25px;
    font-size: 30px;
    cursor: pointer;
    background: none;
    color: #FFF;
    border: none;
}

.dialogTitle {
    font-size: 40px;
    font-weight: 300;
    text-transform: uppercase;
    margin-top: 10px;
}

.confirmDeleteMovieButton {
    background: #F65261;
    text-transform: uppercase;
    font-size: 20px;
    font-weight: 500;
    color: #FFFFFF;
    border: none;
    padding: 10px 35px;
    align-self: flex-end;
    cursor: pointer;
}

.deleteMovieModal {
    display: flex;
}

.deleteMessage {
    font-size: 20px;
}
