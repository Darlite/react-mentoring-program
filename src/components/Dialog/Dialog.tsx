import styles from "./Dialog.module.css";
import React, {ReactNode} from "react";
import {FocusTrap} from "focus-trap-react";

interface DialogProps {
    dialogTitle: string;
    content: ReactNode;
    handleToggleDialog: () => void;
}

const Dialog: React.FC<DialogProps> = ({dialogTitle, content, handleToggleDialog}) => {
    const isDeleteDialog = dialogTitle === "Delete movie";

    return (
        <FocusTrap focusTrapOptions={{}}>
            {<div className={styles.dialogContainer}>
                <h1 className={styles.dialogTitle}>{dialogTitle}</h1>
                {isDeleteDialog ? (
                    <>
                        <p className={styles.deleteMessage}>
                            Are you sure you want to delete this movie?
                        </p>
                        <button className={styles.confirmDeleteMovieButton}>
                            Confirm
                        </button>
                    </>
                ) : (content)
                }
                <button className={styles.closeButton}
                        name="close"
                        aria-label="close"
                        onClick={handleToggleDialog}>X</button>
            </div>
            }
        </FocusTrap>
    )
}

export default Dialog;
