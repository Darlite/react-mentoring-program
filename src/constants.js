export const genreNames = ['All', 'Documentary', 'Comedy', 'Horror', 'Crime'];