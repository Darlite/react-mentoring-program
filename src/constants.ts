export const genreNames: string[] = [
    'All',
    'Documentary',
    'Comedy',
    'Horror',
    'Crime'
];