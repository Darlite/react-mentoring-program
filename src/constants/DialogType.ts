export enum DialogType {
    AddMovie = "ADD MOVIE",
    EditMovie = "EDIT MOVIE",
    DeleteMovie = "DELETE MOVIE",
}
