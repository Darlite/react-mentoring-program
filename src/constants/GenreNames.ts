export const GenreNames: string[] = [
    'All',
    'Documentary',
    'Comedy',
    'Horror',
    'Crime'
];
