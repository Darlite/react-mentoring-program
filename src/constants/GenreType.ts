export enum GenreType {
    Action = "Action",
    Adventure = "Adventure",
    Comedy = "Comedy",
    Crime = "Crime",
    Documentary = "Documentary",
    Drama = "Drama",
    Fantasy = "Fantasy",
    Horror = "Horror",
    Musical = "Musical",
    Mystery = "Mystery",
    Romance = "Romance",
    ScienceFiction = "Science Fiction",
    Thriller = "Thriller",
    War = "War",
}