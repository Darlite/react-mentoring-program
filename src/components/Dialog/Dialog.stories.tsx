import {Meta, StoryObj} from "@storybook/react";
import {fn} from "@storybook/test";
import Dialog from "./Dialog";
import MovieForm from "../MovieForm/MovieForm";
import React from "react";
import {mockMoviesData} from "../../mocks/mockMoviesData";
import {DialogType} from "../../constants/DialogType";
import ModalContent from "../ModalContent/ModalContent";

const meta = {
    component: Dialog,
    title: 'Dialog',
    tags: ['autodocs'],
} satisfies Meta<typeof Dialog>;

export default meta;

type Story = StoryObj<typeof meta>;

export const AddMovie: Story = {
    args: {
        dialogTitle: DialogType.AddMovie,
        content: <MovieForm handleSubmit={fn()}/>,
        handleToggleDialog: fn(),
        showDialog: true,
    },
};

export const EditMovie: Story = {
    args: {
        dialogTitle: DialogType.EditMovie,
        content: <MovieForm initialMovieInfo={mockMoviesData} handleSubmit={fn()}/>,
        handleToggleDialog: fn(),
        showDialog: true,
    },
};

export const DeleteMovie: Story = {
    args: {
        dialogTitle: DialogType.DeleteMovie,
        content: <ModalContent currentDialog={DialogType.DeleteMovie}
                               handleSubmit={fn()} /> ,
        handleToggleDialog: fn(),
        showDialog: true,
    },
};
