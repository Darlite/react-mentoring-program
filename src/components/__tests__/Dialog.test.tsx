import Dialog from "../Dialog/Dialog";
import React from "react";
import {render, screen} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import {DialogType} from "../../constants/DialogType";

describe("Dialog component", () => {
    it("renders correctly for Delete Dialog", () => {
        const { container } = render(<Dialog dialogTitle={DialogType.DeleteMovie}
                                             content={""}
                                             handleToggleDialog={() => {}}
        />);
        expect(container).toMatchSnapshot();
    });

    it("renders Add movie dialog", () => {
        render(<Dialog dialogTitle={DialogType.AddMovie}
                       content={""}
                       handleToggleDialog={() => {}}
        />);

        expect(screen.getByText(DialogType.AddMovie)).toBeInTheDocument();
    });

    it("renders Edit movie dialog", () => {
        render(<Dialog dialogTitle={DialogType.EditMovie}
                       content={""}
                       handleToggleDialog={() => {}}
        />);

        expect(screen.getByText(DialogType.EditMovie)).toBeInTheDocument();
    });

    it("handles the click on the close button", () => {
        const handleToggleDialog = jest.fn();

        render(<Dialog dialogTitle={DialogType.EditMovie}
                       content={""}
                       handleToggleDialog={handleToggleDialog}
        />);

        const closeButton = screen.getByText("X");
        userEvent.click(closeButton);
        expect(handleToggleDialog).toHaveBeenCalled();
    })
})
