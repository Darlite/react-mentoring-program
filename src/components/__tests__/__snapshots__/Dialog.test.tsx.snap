// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component renders correctly for Delete Dialog 1`] = `
<div>
  <div
    class="dialogContainer"
  >
    <h1
      class="dialogTitle"
    >
      DELETE MOVIE
    </h1>
    <p
      class="deleteMessage"
    >
      Are you sure you want to delete this movie?
    </p>
    <button
      aria-label="Confirm"
      class="confirmDeleteMovieButton"
    >
      Confirm
    </button>
    <button
      aria-label="close"
      class="closeButton"
      name="close"
    >
      X
    </button>
  </div>
</div>
`;
